Tests/SwiftLintFrameworkTests/ConfigurationTests+MultipleConfigs.swift

```diff
@@ -301,6 +301,113 @@ extension ConfigurationTests {
         )
     }
 
+    func testParentChildOptInAndDisable() {
+        struct TestCase: Equatable {
+            let optedInInParent: Bool
+            let disabledInParent: Bool
+            let optedInInChild: Bool
+            let disabledInChild: Bool
+            let isEnabled: Bool
+        }
+        let testCases: [TestCase] = [
+            // swiftlint:disable line_length
+            TestCase(optedInInParent: false, disabledInParent: false, optedInInChild: false, disabledInChild: false, isEnabled: false),
+            TestCase(optedInInParent: true, disabledInParent: false, optedInInChild: false, disabledInChild: false, isEnabled: true),
+            TestCase(optedInInParent: false, disabledInParent: true, optedInInChild: false, disabledInChild: false, isEnabled: false),
+            TestCase(optedInInParent: true, disabledInParent: true, optedInInChild: false, disabledInChild: false, isEnabled: false),
+            TestCase(optedInInParent: false, disabledInParent: false, optedInInChild: true, disabledInChild: false, isEnabled: true),
+            TestCase(optedInInParent: true, disabledInParent: false, optedInInChild: true, disabledInChild: false, isEnabled: true),
+            TestCase(optedInInParent: false, disabledInParent: true, optedInInChild: true, disabledInChild: false, isEnabled: true),
+            TestCase(optedInInParent: true, disabledInParent: true, optedInInChild: true, disabledInChild: false, isEnabled: true),
+            TestCase(optedInInParent: false, disabledInParent: false, optedInInChild: false, disabledInChild: true, isEnabled: false),
+            TestCase(optedInInParent: true, disabledInParent: false, optedInInChild: false, disabledInChild: true, isEnabled: false),
+            TestCase(optedInInParent: false, disabledInParent: true, optedInInChild: false, disabledInChild: true, isEnabled: false),
+            TestCase(optedInInParent: true, disabledInParent: true, optedInInChild: false, disabledInChild: true, isEnabled: false),
+            TestCase(optedInInParent: false, disabledInParent: false, optedInInChild: true, disabledInChild: true, isEnabled: false),
+            TestCase(optedInInParent: true, disabledInParent: false, optedInInChild: true, disabledInChild: true, isEnabled: false),
+            TestCase(optedInInParent: false, disabledInParent: true, optedInInChild: true, disabledInChild: true, isEnabled: false),
+            TestCase(optedInInParent: true, disabledInParent: true, optedInInChild: true, disabledInChild: true, isEnabled: false)
+            // swiftlint:enable line_length
+        ]
+        XCTAssertEqual(testCases.unique.count, 4 * 4)
+        let ruleType = ImplicitReturnRule.self
```

Comment:
It's probably helpful to assert that this exemplary rule is an `OptInRule` in case that changes at some time later and suddenly the test fails.

Same for the other test case where the rule is active by default.